README.md

```diff
@@ -47,6 +45,8 @@ The code has been compiled and tested with
 clang version 7.0 and 8.0
 using either GNU's libstdc++ or LLVM's libc++.
 
+
+## Building Flang out of tree
```

Comment:
I think I would put a little intro here saying what these instructions are for and why you might need to follow them. Probably also recommend building in-tree as far easier.